### PR TITLE
Applying security patches to urllib3 and sqlparse

### PR DESCRIPTION
**Pull request by casey-williams-rh**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-15127
- https://issues.redhat.com/browse/RHCLOUD-16336

## Description of Intent of Change(s)
The following code modules have been updated to mitigate `CVE-2021-33503` and `CVE-2021-32839`.

- Updated `sqlparse` from "0.3.0" to "0.4.2"
- Updated `urllib3` from "1.25.9" to "1.26.6"
- Updated `watchtower` from "0.6.0" to "1.0.6"
- Updated `boto3` from "1.9.216" to "1.18.49"

**Please note:** A number of sub-dependencies tied to the dependencies listed above have also been updated as a byproduct of this security update (_Detailed in pipfile.lock_). 

## Local Testing
- Local API Testing == ✅. Passed
- TOX Testing == ✅. Passed

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [x] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [X] General Coding Practices

### CODE REVIEW

File: docs/rtd_requirements.txt [hunk 1]
```diff
@@ -1,41 +1,57 @@
 -i https://pypi.python.org/simple
-amqp==2.5.1
-billiard==3.6.1.0
-boto3==1.9.216
-botocore==1.12.216
+amqp==2.6.1
```
Comment (astrozzc) [inline docs/rtd_requirements.txt:2]: Overall LGTM, just curious, I haven't updated this file before, do you know why this file is updated?

File: Pipfile [hunk 2]
```diff
@@ -17,13 +17,13 @@ djangorestframework-csv = "==2.1.0"
 pytz = "==2019.2"
 django-prometheus = "==1.0.15"
 prometheus-client = "==0.7.1"
-urllib3 = "==1.25.9"
+urllib3 = "==1.26.6"
 Jinja2 = "==2.11.3"
-watchtower = "==0.6.0"
-boto3 = "==1.9.216"
+watchtower = "==1.0.6"
```
Comment (coderbydesign) [inline Pipfile:22]: Looks like `watchtower` is a major release bump. Do you know what breaking changes may have been introduced with v1?

File: docs/rtd_requirements.txt (hunk 1 above)
Comment (casey-williams-rh) [reply docs/rtd_requirements.txt:2]: `docs/rtd_requirements.txt` is the legacy requirements file that is not used in the build of RBAC anymore due to the use of `pipenv`. I have been keeping it updated so Synk can scan the dependencies correctly, as it cannot currently scan `Pipfile` or `Pipfile.lock`

File: Pipfile (hunk 2 above)
Comment (casey-williams-rh) [reply Pipfile:22]: @coderbydesign , judging by the [change log](https://github.com/kislyuk/watchtower/blob/5002639093154d29fdeb1bbeb8a2fb30c3b7f48e/Changes.rst) for `watchtower`, the updates consisted of a bug fixes, a couple of new functions/features, and the drop of Python 2 support. I don't believe there any breaking changes.

**Review by astrozzc — APPROVED** [review]
LGTM, up to Keith and Chris

**Pull request merged**